lib/src/main/java/org/altbeacon/beacon/BeaconManager.java

```diff
@@ -617,6 +621,10 @@ public long getForegroundBetweenScanPeriod() {
         return foregroundBetweenScanPeriod;
     }
 
+    public void setBackgroundScanJobFirstRun(Boolean BackgroundScanJobFirstRun) {
```

Comment:
Can we put this flag on ScanJobScheduler instead of BeaconManager?  I would prefer to not put it on BeaconManager since this is a public interface for the library and this flag should not be part of that public interface.  If you move the flag to ScanJobScheduler, then BeaconManager can hold a private reference to the ScanJobScheduler when it uses it, so we make sure we are always using the same instance of ScanJobScheduler.  This way it is logically equivalent to the code you have.

Reply:
Definitely makes more sense to be like you suggested. I've removed the flag from _BeaconManager_ and _ScanState_ class. Everything is no handled inside _ScanJobScheduler_ class.